src/main/comments.js

```diff
@@ -540,23 +540,18 @@ function printDanglingComments(path, options, sameIndent, filter) {
   return indent([hardline, join(hardline, parts)]);
 }
 
-function prependCursorPlaceholder(path, options, printed) {
-  if (path.getNode() === options.cursorNode && path.getValue()) {
-    return [cursor, printed, cursor];
-  }
-  return printed;
-}
-
-function printComments(path, printed, options, needsSemi) {
+function printCommentsSeparately(path, options) {
   const value = path.getValue();
-  const comments = value && value.comments;
+  const hasComments = isNonEmptyArray(value && value.comments);
+  const isCursorNode = Boolean(path.getNode() === options.cursorNode && value);
```

Comment:
Why the old `prependCursorPlaceholder` use both `path.getNode` and `path.getValue`? Can we use `value` here?

Reply:
No idea. I think we can.